packages/web3-common/src/eth_execution_api.ts

```diff
@@ -0,0 +1,266 @@
+	readonly receiptsRoot: HexString32Bytes;
+	readonly logsBloom: HexString256Bytes;
+	readonly difficulty?: HexStringBytes;
+	readonly number: Uint;
+	readonly gasLimit: Uint;
+	readonly gasUsed: Uint;
+	readonly timestamp: Uint;
+	readonly extraData: HexStringBytes;
+	readonly mixHash: HexString32Bytes;
+	readonly nonce: HexStringBytes;
+	readonly totalDifficulty: Uint;
+	readonly baseFeePerGas?: Uint;
+	readonly size: Uint;
+	readonly transactions: TransactionHash[] | TransactionSigned[];
+	readonly uncles: Uncles;
+}
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/filter.json#L59
+export type Topic = HexString256Bytes;
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/receipt.json#L2
+export interface Log {
+	readonly removed?: boolean;
+	readonly logIndex?: Uint;
+	readonly transactionIndex?: Uint;
+	readonly transactionHash?: HexString32Bytes;
+	readonly blockHash?: HexString32Bytes;
+	readonly blockNumber?: Uint;
+	readonly address?: Address;
+	readonly data?: HexStringBytes;
+	readonly topics?: Topic[];
+}
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/receipt.json#L44
+export interface ReceiptInfo {
+	readonly transactionHash: HexString32Bytes;
+	readonly transactionIndex: HexString32Bytes;
+	readonly blockHash?: HexString32Bytes;
+	readonly blockNumber?: Uint;
+	readonly from?: Address;
+	readonly to?: Address;
+	readonly cumulativeGasUsed?: Uint;
+	readonly gasUsed?: Uint;
+	readonly contractAddress?: Address;
+	readonly logs?: Log[];
+	readonly logsBloom?: HexString256Bytes;
+}
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/client.json#L2
+export type SyncingStatus =
+	| { startingBlock: Uint; currentBlock: Uint; highestBlock: Uint }
+	| boolean;
+
+// https://github.com/ethereum/execution-apis/blob/main/src/eth/fee_market.json#L53
+export interface FeeHistoryResult {
+	readonly oldestBlock: Uint;
+	readonly baseFeePerGas: Uint;
+	readonly reward: number[][];
+}
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/filter.json#L28
+export interface Filter {
+	readonly fromBlock?: Uint;
+	readonly toBlock?: Uint;
+	readonly address?: Address | Address[];
+	readonly topics?: Topic[];
+}
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/filter.json#L2
+export type FilterResults = HexString32Bytes[] | Log[];
+
+/* eslint-disable camelcase */
+export type EthExecutionAPI = {
+	// https://github.com/ethereum/execution-apis/blob/main/src/eth/block.json
+	eth_getBlockByHash: (blockHash: HexString32Bytes, hydrated: boolean) => Block;
+	eth_getBlockByNumber: (blockNumber: Uint, hydrated: boolean) => Block;
+	eth_getBlockTransactionCountByHash: (blockHash: HexString32Bytes) => Uint[];
+	eth_getBlockTransactionCountByNumber: (blockNumber: Uint) => Uint[];
+	eth_getUncleCountByBlockHash: (blockHash: HexString32Bytes) => Uint[];
+	eth_getUncleCountByBlockNumber: (blockNumber: Uint) => Uint[];
```

Comment:
```suggestion
	eth_getBlockTransactionCountByHash: (blockHash: HexString32Bytes) => Uint;
	eth_getBlockTransactionCountByNumber: (blockNumber: Uint) => Uint;
	eth_getUncleCountByBlockHash: (blockHash: HexString32Bytes) => Uint;
	eth_getUncleCountByBlockNumber: (blockNumber: Uint) => Uint;
```

These are returned as hex strings

Reply:
I could not find `eth_getUncleByBlockHashAndIndex` and `eth_getUncleByBlockNumberAndIndex` anywhere in the specification. Can you please suggest some reference.

Reply:
I created [an issue](https://github.com/ethereum/execution-apis/issues/96) to ask about the return types

Reply:
https://github.com/ethereum/execution-apis/issues/30

Reply:
https://github.com/ethereum/execution-apis/issues/102